packages/api/src/ApiRx.ts

```diff
@@ -23,17 +23,29 @@ import {ApiOptions as ApiOptionsBase} from '@plugnet/api/types';
 import {ProviderInterface} from '@plugnet/rpc-provider/types';
 import {isFunction, isObject} from '@plugnet/util';
 
-import {Observable} from 'rxjs';
+import {fromEvent, Observable, race, throwError} from 'rxjs';
+import {switchMap, timeout} from 'rxjs/operators';
+import {DEFAULT_TIMEOUT} from './Api';
 import * as derives from './derives';
 import staticMetadata from './staticMetadata';
 import {ApiOptions, IPlugin} from './types';
 import {getProvider} from './util/getProvider';
+import {getTimeout} from './util/getTimeout';
 import logger from './util/logging';
+
 const Types = require('@cennznet/types');
 
 export class ApiRx extends ApiRxBase {
     static create(options: ApiOptions | ProviderInterface = {}): Observable<ApiRx> {
-        return new ApiRx(options).isReady;
+        const apiRx = new ApiRx(options);
+
+        const timeoutMs = getTimeout(options);
+        return race(
+            apiRx.isReady.pipe(timeout(timeoutMs ? timeoutMs : DEFAULT_TIMEOUT)),
```

Comment:
1. from what i tested, if 0 or neg number is passed to timeout, timeout will always be triggered. which is not what we want here.
2.  don't use `timeoutMs ?` to check undefined, because timeoutMs is a number, 0 will be reckon as false.

Reply:
It's expected that timeout will be triggered if neg number is passed to timeout. And timeout should not be triggered if 0 is passed in, I tested [here](https://github.com/cennznet/api.js/blob/bf3d189272332f79829b1773eb5df7ed78d52ece/packages/api/test/e2e/rxapi.create.e2e.ts#L28)

Reply:
Ah right, will update this. I tested [here](https://github.com/cennznet/api.js/blob/bf3d189272332f79829b1773eb5df7ed78d52ece/packages/api/test/e2e/rxapi.create.e2e.ts#L28), looks like this case is passed because of using the DEFAULT_TIMEOUT when 0 is passed in...